amqpdispatcher/amqp_proxy.py

```diff
@@ -1,41 +1,59 @@
 #!/usr/bin/env python
 # -*- coding:utf-8 -*-
+from typing import Dict, Any
 
-from amqpdispatcher.channel_proxy import proxy_channel
+from aio_pika import Channel, Exchange, Connection
+from aio_pika import Message as AioPikaMessage
 
+from amqpdispatcher.message import Message
 
-class AMQPProxy(object):
 
-    def __init__(self, channel, msg):
-        self._channel = proxy_channel(channel)
-        self._msg = msg
+class AMQPProxy(object):
+    _connection: Connection
+    _publish_channel: Channel
+    _terminal_state: bool
+    _message: Message
+
+    def __init__(self, connection: Connection, channel: Channel, message: Message):
+        self._message = message
+        self._publish_channel = channel
+        self._connection = connection
         self._terminal_state = False
 
     @property
-    def tag(self):
-        return self._msg.delivery_info['delivery_tag']
-
-    @property
-    def has_responded_to_message(self):
+    def has_responded_to_message(self) -> bool:
         return self._terminal_state
 
-    def ack(self):
+    async def ack(self) -> None:
         self._error_if_already_terminated()
-        self._channel.basic_ack(self.tag)
+        await self._message.raw_message.ack()
 
-    def nack(self):
+    async def nack(self) -> None:
         self._error_if_already_terminated()
-        self._channel.basic_nack(self.tag)
+        await self._message.raw_message.nack()
 
-    def reject(self, requeue=True):
+    async def reject(self, requeue: bool = True) -> None:
         self._error_if_already_terminated()
-        self._channel.basic_reject(self.tag, requeue=requeue)
-
-    def publish(self, exchange, routing_key, headers, body):
-        self._channel.basic_publish(exchange, routing_key, body, headers)
-
-    def _error_if_already_terminated(self):
+        await self._message.raw_message.reject(requeue=requeue)
+
+    async def publish(
+        self, exchange_name: str, routing_key: str, headers: Dict[Any, Any], body: bytes
```

Comment:
shouldn't the headers be `Dict[str, str]`?

Reply:
Just looked at the first signature for `headers` and it appears to expect `Dict[str, Any]`, so changing to that.